common/src/Microsoft.Azure.IIoT.Hub.Module.Client/src/Default/IoTSdkFactory.cs

```diff
@@ -205,471 +225,33 @@ public sealed class IoTSdkFactory : IClientFactory, IDisposable {
-
-                if (status == ConnectionStatus.Connected) {
-                    logger.Information("{counter}: Device {deviceId} reconnected " +
-                        "due to {reason}.", _reconnectCounter, deviceId, reason);
-                    kReconnectionStatus.WithLabels(deviceId).Set(_reconnectCounter);
-                    _reconnectCounter++;
-                    return;
-                }
-                logger.Information("{counter}: Device {deviceId} disconnected " +
-                    "due to {reason} - now {status}...", _reconnectCounter, deviceId,
-                        reason, status);
-                kDisconnectionStatus.WithLabels(deviceId).Set(_reconnectCounter);
-                if (IsClosed) {
-                    // Already closed - nothing to do
-                    return;
-                }
-                if (status == ConnectionStatus.Disconnected ||
-                    status == ConnectionStatus.Disabled) {
-                    // Force
-                    IsClosed = true;
-                    onConnectionLost?.Invoke();
-                }
-            }
-
-            /// <summary>
-            /// Helper to create device client
-            /// </summary>
-            /// <param name="cs"></param>
-            /// <param name="transportSetting"></param>
-            /// <returns></returns>
-            private static DeviceClient Create(IotHubConnectionStringBuilder cs,
-                ITransportSettings transportSetting) {
-                if (cs == null) {
-                    throw new ArgumentNullException(nameof(cs));
-                }
-                if (transportSetting != null) {
-                    return DeviceClient.CreateFromConnectionString(cs.ToString(),
-                        new ITransportSettings[] { transportSetting });
+                    throw new InvalidConfigurationException(
+                        "No connection string for device client specified.");
                 }
-                return DeviceClient.CreateFromConnectionString(cs.ToString());
             }
-
-            private readonly DeviceClient _client;
-            private readonly ILogger _logger;
-            private int _reconnectCounter;
-            private static readonly Gauge kReconnectionStatus = Metrics
-                .CreateGauge("iiot_edge_device_reconnected", "reconnected count",
-                    new GaugeConfiguration {
-                        LabelNames = new[] { "device"}
-                    });
-            private static readonly Gauge kDisconnectionStatus = Metrics
-                .CreateGauge("iiot_edge_device_disconnected", "disconnected count",
-                    new GaugeConfiguration {
-                        LabelNames = new[] { "device"}
-                    });
+            return ModuleClientAdapter.CreateAsync(product, _deviceClientCs, DeviceId, ModuleId,
+                transportSetting, timeout: TimeSpan.FromMinutes(5), RetryPolicy, onError, _logger);
```

Comment:
again the 5 minutes magic timeout

Reply:
Change to _timeout variable but that one needs to come from configuration.